src/internal/types.ts

```diff
@@ -96,15 +97,25 @@ type IsBoundedString<T> = T extends string
  *
  * @see EnumerableStringKeyedValueOf
  */
-export type EnumerableStringKeyOf<T> = `${Exclude<keyof T, symbol>}`;
+export type EnumerableStringKeyOf<T> =
+  T extends Record<infer K, unknown> ? `${Exclude<K, symbol>}` : never;
+
+/**
+ * Extracts the value type from an object type T.
+ */
+export type InferredRecordValue<T> = T extends EmptyObject
+  ? T[keyof T]
```

Comment:
**question:** why is there a `T extends EmptyObject` check here? what goes wrong without it?

can you add a comment for this

Reply:
This was added mainly due to a failing test case for `values`: 
```ts
it("should correctly type functions", () => {
    const result = values(doNothing());
    expectTypeOf(result).toEqualTypeOf<Array<never>>();
});
```
Without the check for `EmptyObject`, result is typed as `uknown[]` instead of `never[]`. I agree that it is a bit confusing and a comment would be useful. If you have a better suggestion, I would be happy to integrate it; otherwise, I will proceed with adding a comment.

Reply:
makes sense, let's comment